Tests/NIOTests/BootstrapTest.swift

```diff
@@ -205,4 +205,38 @@ class BootstrapTest: XCTestCase {
         XCTAssertNoThrow(try childChannelDone.futureResult.wait())
         XCTAssertNoThrow(try serverChannelDone.futureResult.wait())
     }
+
+    func testClientTransportBootstrapAllowsConformanceCorrectly() throws {
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer {
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+        let clientBootstrap: NIOClientTransportBootstrap = ClientBootstrap(group: group)
```

Comment:
It might be nicer to write a function that takes the bootstrap generically and write the test in that function. That way it's much harder to accidentally break this test by changing only one line in a seemingly-innocuous way.